src/app/js/admin/field/FieldAddDropdownButton.js

```diff
@@ -21,27 +21,29 @@ import { fromFields } from '../../sharedSelectors';
 import { addField } from '../../fields';
 import { polyglot as polyglotPropTypes } from '../../propTypes';
 import { SCOPE_DOCUMENT } from '../../../../common/scope';
+import { useParams } from 'react-router';
 
 const options = [
     { label: 'new_field', icon: <Add /> },
     { label: 'blank_field', icon: <Add /> },
     { label: 'from_original_dataset', icon: <AddFromDatasetIcon /> },
 ];
 
-const FieldAddDropdownButton = ({
+export const FieldAddDropdownButtonComponent = ({
     onAddNewField,
     onShowExistingColumns,
     isFieldsLoading,
     p: polyglot,
 }) => {
     const [open, setOpen] = React.useState(false);
+    const { subresourceId } = useParams();
```

Comment:
This component could remain a dumb component if `subresourceId` was handled by one of the parent components.